index.js

```diff
@@ -85,263 +86,251 @@ class LatticeKeyring extends EventEmitter {
-            value: `0x${tx.value.toString('hex')}`,
-            data: tx.data.length === 0 ? null : `0x${tx.data.toString('hex')}`,
-            signerPath: this._getHDPathIndices(addressParentPath, addressIdx),
-          }
-          switch (tx._type) {
-            case 2: // eip1559
-              if ((tx.maxPriorityFeePerGas === null || tx.maxFeePerGas === null) ||
-                  (tx.maxPriorityFeePerGas === undefined || tx.maxFeePerGas === undefined))
-                throw new Error('`maxPriorityFeePerGas` and `maxFeePerGas` must be included for EIP1559 transactions.');
-              txData.maxPriorityFeePerGas = `0x${tx.maxPriorityFeePerGas.toString('hex')}`;
-              txData.maxFeePerGas = `0x${tx.maxFeePerGas.toString('hex')}`;
-              txData.accessList = tx.accessList || [];
-              txData.type = 2;
-              break;
-            case 1: // eip2930
-              txData.accessList = tx.accessList || [];
-              txData.gasPrice = `0x${tx.gasPrice.toString('hex')}`;
-              txData.type = 1;
-              break;
-            default: // legacy
-              txData.gasPrice = `0x${tx.gasPrice.toString('hex')}`;
-              txData.type = null;
-              break;
-          }
-          // Lattice firmware v0.11.0 implemented EIP1559 and EIP2930 so for previous verisons
-          // we need to overwrite relevant params and revert to legacy type.
-          // Note: `this.sdkSession.fwVersion is of format [fix, minor, major, reserved]
-          const forceLegacyTx = this.sdkSession.fwVersion[2] < 1 && 
-                                this.sdkSession.fwVersion[1] < 11;
-          if (forceLegacyTx && txData.type === 2) {
-            txData.gasPrice = txData.maxFeePerGas;
-            txData.revertToLegacy = true;
-            delete txData.type;
-            delete txData.maxFeePerGas;
-            delete txData.maxPriorityFeePerGas;
-            delete txData.accessList;
-          } else if (forceLegacyTx && txData.type === 1) {
-            txData.revertToLegacy = true;
-            delete txData.type;
-            delete txData.accessList;
-          }
-          // Get the signature
-          return this._signTxData(txData)
-        } catch (err) {
-          throw new Error(`Failed to build transaction.`)
+  async getAccounts() {
+    return this.accounts ? this.accounts.slice() : [].slice();
+  }
+
+  async signTransaction (address, tx) {
+    let signedTx, v;
+    const txToReturn = tx.toJSON();
+    txToReturn.type = tx._type || null;
+    const accountIdx = await this._findSignerIdx(address);
+    const chainId = getTxChainId(tx).toNumber();
+    const fwVersion = this.sdkSession.getFwVersion();
+    // Build the Lattice request data and make request
+    // We expect `tx` to be an `ethereumjs-tx` object, meaning all fields are bufferized
+    // To ensure everything plays nicely with gridplus-sdk, we convert everything to hex strings
```

Comment:
We could remove this comment

Reply:
[52b13a0](https://github.com/GridPlus/eth-lattice-keyring/pull/33/commits/52b13a02e9654d2b9b693627bc746868dd62e980)